gradle-plugin/generate_build_properties.gradle

```diff
@@ -32,4 +33,13 @@ def generateBuildProperties = project.tasks.register('generateBuildProperties')
 
 compileKotlin {
   dependsOn generateBuildProperties
-}
\ No newline at end of file
+}
+
+// Add the task dependencies to enforce correct ordering when depending out outputs.
+tasks.whenTaskAdded {
+  // For some reason when using `tasks.named('sourcesJar')` Gradle complains that the task cannot
+  // be found. Probably a lifecycle issue with afterEvaluate...
```

Comment:
I was gonna ask why you chose to do it this way lol. Does dropping an `afterEvaluate` in here still not resolve the issue?

Reply:
Oh, that would work, too. Not sure which one is better now ¯\\_(ツ)_/¯

Reply:
`afterEvaluate` + `named` would be more...targetted. You would only run that lambda twice in total instead of for each task registered.

That would be _preferred_, but if it still doesn't work I wouldn't lose sleep over this approach either.

Reply:
Changed. `named` also works with lazy APIs.